.github/workflows/test.yml

```diff
@@ -37,17 +37,17 @@ jobs:
           submodules: true
 
       - name: Install Nix
-        uses: cachix/install-nix-action@v16
+        uses: cachix/install-nix-action@v31
```

Comment:
Unrelated to the bulk of this PR, but I wonder if we might want to start using Dependabot on this repo too so that we don't get so far out-of-date, and so that we don't have to manually bump these numbers. IMO it's been working fine in p-u since https://github.com/GaloisInc/parameterized-utils/pull/195, see e.g., https://github.com/GaloisInc/parameterized-utils/pulls?q=is%3Apr+is%3Aclosed+author%3Aapp%2Fdependabot.